src/pushcollector/_impl/collector.py

```diff
@@ -152,5 +160,13 @@ def _require_backend(cls, name):
             raise ValueError("No registered pushcollector backend: '%s'" % name)
 
 
+def context_managed_collector(fn):
```

Comment:
Not sure I get the purpose of this function.  If you want to add it to public API, it'll need docs.  I'm doubtful it's needed though, my suggestion is to leave it out.

You might enjoy https://url.corp.redhat.com/3f5abee , https://url.corp.redhat.com/aaf71c9 , "When in doubt, leave it out" is one of the catchy API design guidelines from that presentation.

Reply:
Ill check those out, thanks!

My motivation for this function was to make integrating these changes easier. For example, in `rcm-pub` the Push class has at least 3 methods that I could see that would want to make use of these new context managed Collector changes. Rather than doing something like this:
```python
def do_push(self):
    with Collector.get():
        # existing code gets bulk indented one more level
```

in each of those methods, I wanted something like this instead:
```python
@context_managed_collector
def do_push(self):
    # existing code is left alone
```

which is less invasive, and gets the same benefit.

The decorator could even be updated to take an argument that specifies the backend the client wants to use so that you get:
```python
@context_managed_collector("local")
def do_push(self):
    ...
```

which I think looks nice. Happy to drop it if you want, but I definitely will add docs to it if we decide to keep it.

Reply:
While I think the decorator is OK, I also think it'd be better to implement it directly in Pub which I suppose will be the only user of it for the time being.  In general, if someone outside of a library could easily implement some convenience method themselves using only the public API (which is true here) then I think that's the right thing to do.  It should only be added into the library once we have a demonstrated need for more than one user.  Otherwise, libraries tend to grow all kinds of ad-hoc convenience methods just because someone needed it once.

Reply:
Okay, I think that makes sense. I will move it over to Pub and remove it from here.